pkg/kubectl/cmd/run.go

```diff
@@ -213,7 +214,9 @@ func Run(f cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *cobr
 		}
 		switch restartPolicy {
 		case api.RestartPolicyAlways:
-			if contains(resourcesList, v1beta1.SchemeGroupVersion.WithResource("deployments")) {
+			if contains(resourcesList, appsv1beta1.SchemeGroupVersion.WithResource("deployments")) {
+				generatorName = "deployment/apps.v1beta1"
+			} else if contains(resourcesList, extensionsv1beta1.SchemeGroupVersion.WithResource("deployments")) {
 				generatorName = "deployment/v1beta1"
```

Comment:
Changing default generator is breaking. 

We should use `deployment/v1beta1` by default. 
If the users specified `deployment/apps.v1beta1`, we discover if deployment.apps is supported first. If so, create deployment.apps; otherwise, create deployment.extensions, but manually add those new defaults. 

We need to test these cases too.